pkgs/applications/editors/neovim/wrapper.nix

```diff
@@ -29,25 +29,25 @@ let
     '';
   };
 
+  /* for compatibility with passing extraPythonPackages as a list; added 2018-07-11 */
+  compatFun = funOrList: (if builtins.isList funOrList then
+    (_: builtins.trace "passing a list as extraPythonPackages to the neovim wrapper is deprecated, pass a function as to python.withPackages instead" funOrList)
+    else funOrList);
+  extraPythonPackagesFun = compatFun extraPythonPackages;
+  extraPython3PackagesFun = compatFun extraPython3Packages;
+
   pluginPythonPackages = if configure == null then [] else builtins.concatLists
     (map ({ pythonDependencies ? [], ...}: pythonDependencies)
          (vimUtils.requiredPlugins configure));
-  pythonEnv = pythonPackages.python.buildEnv.override {
-    extraLibs = (
-        if withPyGUI
-          then [ pythonPackages.neovim_gui ]
-          else [ pythonPackages.neovim ]
-      ) ++ extraPythonPackages ++ pluginPythonPackages;
-    ignoreCollisions = true;
-  };
+  pythonEnv = pythonPackages.python.withPackages(ps:
+        (if withPyGUI then [ ps.neovim_gui ] else [ ps.neovim ])
+        ++ (extraPythonPackagesFun ps) ++ pluginPythonPackages);
```

Comment:
This will not work yet with `pluginPythonPackages` since `ps` is not passed to `pluginPythonPackages`. pluginPythonPackages also does not work with vim's own package manager. We only have one plugin that is using this mechanism - should it be removed for the moment? It is also not implemented for vim8.

Reply:
Hum I think it still works, but indeed adding a plugin from stable will not work with neovim from unstable. I'll try to adapt it.

Reply:
Hum does anyone have an example of how to use the configure argument of the wrapper ?

Reply:
I'm using it here: https://github.com/manveru/vimrc/blob/master/default.nix